File: api/src/main/java/com/FireSale/api/dto/user/UserDTO.java
package com.FireSale.api.dto.user;

import com.FireSale.api.dto.BaseDTO;
import com.FireSale.api.model.Gender;
import com.FireSale.api.model.Role;
import com.FireSale.api.dto.address.AddressDTO;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDate;

@Data
public class UserDTO extends BaseDTO {
    private String email;
    private String displayName;
    private String firstName;
    private String lastName;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateOfBirth;
    private Gender gender;
    private Role role;
    private AddressDTO address;
    private AddressDTO shippingAddress;
    private byte[] avatar;
}

File: client/FireSale/src/app/shared/models/userDto.ts
import { BaseDto as BaseDTO } from 'src/app/shared/baseDto';
import { AddressDTO } from './addressDto';

export interface UserDTO extends BaseDTO {
  email: string;
  gender: string;
  firstName: string;
  lastName: string;
  address: AddressDTO;
  shippingAddress: AddressDTO;
  dateOfBirth: Date;
  displayName: string;
  password: string;
  passwordVerify: string;
  authData: string;
  avatar: string;
  role: string;
}
